crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -757,60 +717,16 @@ fn try_create_dynamic_field_info(
         return Ok(None);
     }
 
-    let struct_tag: StructTag = move_object.type_().clone().into();
-    let move_struct_layout = struct_tag_to_move_struct_layout
-        .get(&struct_tag)
-        .cloned()
-        .ok_or_else(|| {
-            IndexerError::DynamicFieldError(format!(
-                "Cannot find struct layout in mapfor {:?}.",
-                struct_tag
-            ))
-        })?;
-    let move_struct = move_object.to_move_struct(&move_struct_layout)?;
-    let (move_value, type_, object_id) =
-        DynamicFieldInfo::parse_move_object(&move_struct).tap_err(|e| warn!("{e}"))?;
-    let name_type = move_object.type_().try_extract_field_name(&type_)?;
-    let bcs_name = bcs::to_bytes(&move_value.clone().undecorate()).map_err(|e| {
-        IndexerError::SerdeError(format!(
-            "Failed to serialize dynamic field name {:?}: {e}",
-            move_value
+    let type_: TypeTag = move_object.type_().clone().into();
+    let layout = layouts.get(&type_).ok_or_else(|| {
```

Comment:
wondering if `layouts` is necessary, iirc to get df_kind, a `StructTag` is sufficient?
https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/dynamic_field.rs#L113-L117
https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/dynamic_field.rs#L258-L259

Reply:
Ah yes, good spot, we can avoid deserialization altogether here! I will make that change, thanks.

Reply:
I ended up doing this further simplification in its own PR because it was suitably different and had further implications:

#19622